feat: cooldown cache & database

darksaid98 opened the pull request:
This adds a basic cache for storing cooldowns and support for multiple database drivers so cooldown data can persist between server restarts.

darksaid98 committed:
feat: add cooldown manager class

darksaid98 committed:
fix: clean player cache on reload

darksaid98 committed:
merge: changes from upstream plugin template

darksaid98 committed:
build: add database tooling

darksaid98 committed:
refactor: rename utils package to utility

darksaid98 committed:
feat: add database support

EepyRose commented:
Thanks for this! Is there anything more you want to go over with this or would you consider it ready for merge?

darksaid98 commented:
> Thanks for this! Is there anything more you want to go over with this or would you consider it ready for merge?

I'd prefer it if you continued this PR by making commits on the same branch until this is stable.

The db and cache should work perfectly fine, I've not implemented any usage for them, however.

Note that this is also completely untested so it needs a bit of work.

darksaid98 committed:
feat(db): get cooldown

EepyRose committed:
feat: implemented cooldown for all teleports

darksaid98 committed:
fix(db): date time loading

darksaid98 committed:
feat(commands): reload command functionality

- Makes the reload command reload the plugin
- Still needs to force load player data on startup

darksaid98 commented:
Fixed the DB issue. I removed `hsql` as a suggested DB driver as it was buggy for me and made `h2` the default.

Also made the reload command run the correct code. See the **"todo"** in the relevant commit for some more stuff needed for reload functionality to properly work. https://github.com/Alathra/AlathranCommands/pull/2/commits/63be11581538e795f556e573042715fc3a84cd93

EepyRose committed:
fix: fixed getRemainingCooldownString formatting

EepyRose committed:
fix: made WildLocation#search actually try multiple locations

EepyRose commented:
The latest two commits were supposed to be on the main branch, my bad heh

EepyRose committed:
fix: fixed getRemainingCooldownString formatting (again)

EepyRose committed:
feat: added /clearcooldowns command

darksaid98 reviewed (commented):
Still some pending TODO's in the project. I recommend you search through it and complete all the relevant ones before release.

EepyRose committed:
feat: added grace time to wildTP

also removed already done TODOs

darksaid98 commented on src/main/java/io/github/Alathra/AlathranCommands/data/model/TPARequest.java:
```diff
@@ -98,6 +105,7 @@ public boolean isExpired() {
      *
      * @return the boolean
      */
+    // TODO correct canAfford to reflect if the teleported player can afford
```
Is this done?

EepyRose replied:
This is fixed as of latest commit.

darksaid98 commented on src/main/java/io/github/Alathra/AlathranCommands/runnable/WildTeleportTask.java:
```diff
@@ -0,0 +1,116 @@
+        }
+    }
+
+    void teleport(Player player) {
```
Empty method.

EepyRose replied:
Fixed.

darksaid98 reviewed (commented):
Saw your message on Discord. This just needs final testing on the new test server which will be up in the following days.

EepyRose committed:
fix: corrected TPARequest#canAfford

EepyRose committed:
removed empty method

EepyRose committed:
feat: force loading data for logged on players

reload support
